include/jemalloc/internal/rtree_inlines.h

```diff
@@ -348,7 +348,7 @@ rtree_read(tsdn_t *tsdn, rtree_t *rtree, rtree_ctx_t *rtree_ctx, uintptr_t key,
 	rtree_elm_t *elm;
 
 	elm = rtree_elm_lookup(tsdn, rtree, rtree_ctx, key, dependent, false);
-	if (elm == NULL) {
+	if (!dependent && elm == NULL) {
```

Comment:
Dependent lookups cannot result in ```NULL```, since by definition the rtree element must already exist, so this allows the ```elm == NULL``` check to be omitted at compile time.  In practice this doesn't do us any good anymore, because I recently removed the last uses of ```rtree_read()``` (5177995530557521d330486a3971469e1573d6fc).